dapp/src/lib/utils/buildNameTree.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2025 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+export type NameTree = {
+    name: string;
+    subnames: NameTree[];
+};
+
+export function buildNameTree(name: string, ownedSubnames: string[]): NameTree {
+    const normalizedName = name.toLowerCase();
+    const parts = normalizedName.split('.');
+
+    const children = ownedSubnames
+        .filter((sub) => {
+            if (sub === normalizedName) return false;
+            const subParts = sub.split('.');
+            return (
+                subParts.slice(-parts.length).join('.') === parts.join('.') &&
+                subParts.length === parts.length + 1
+            );
```

Comment:
just to clarify the code
```suggestion
			// Check if the subname has the name as parent, and if the length matches
            return (
                subParts.slice(-parts.length).join('.') === normalizedName &&
                subParts.length === parts.length + 1
            );
```